hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoUpdateHandler.java

```diff
@@ -63,8 +105,169 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
+            builder.key(op.key());
+        }
+        if (op.hasExpirationTime()) {
+            builder.expiry(op.expirationTime().seconds());
+        }
+        if (op.hasReceiverSigRequiredWrapper()) {
+            builder.receiverSigRequired(op.receiverSigRequiredWrapper().booleanValue());
+        } else if (op.hasReceiverSigRequired()) {
+            builder.receiverSigRequired(op.receiverSigRequired());
+        }
+        if (op.hasAutoRenewPeriod()) {
+            builder.autoRenewSecs(op.autoRenewPeriod().seconds());
+        }
+        if (op.hasMemo()) {
+            builder.memo(op.memo());
+        }
+        if (op.hasMaxAutomaticTokenAssociations()) {
+            builder.maxAutoAssociations(op.maxAutomaticTokenAssociations());
+        }
+        if (op.hasDeclineReward()) {
+            builder.declineReward(op.declineReward().booleanValue());
+        }
+        if (op.hasStakedAccountId() || op.hasStakedNodeId()) {
+            builder.stakedNumber(getStakedId(op.stakedId().kind().toString(), op.stakedNodeId(), op.stakedAccountId()));
+        }
+        return builder;
+    }
+
+    /**
+     * Validate semantics of the transaction. This method is called during the handle workflow.
+     * It validates any fields of the transaction that involves state or config.
+     * @param context handle context
+     * @param updateAccount account to be updated
+     * @param op crypto update transaction body
+     * @param builder account update builder
+     * @param accountStore account store
+     */
+    private void validateSemantics(
+            @NonNull final HandleContext context,
+            @NonNull final Account updateAccount,
+            @NonNull final CryptoUpdateTransactionBody op,
+            @NonNull Account.Builder builder,
+            @NonNull final ReadableAccountStore accountStore) {
+        final var expiryValidator = context.expiryValidator();
+        final var builderAccount = builder.build();
+        validateFields(op, context, accountStore);
+
+        validateTrue(!updateAccount.smartContract(), INVALID_ACCOUNT_ID);
+
+        // get needed configs for validation
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var entitiesConfig = context.configuration().getConfigData(EntitiesConfig.class);
+
+        // validate expiry metadata
+        final var currentMetadata = new ExpiryMeta(
+                updateAccount.expiry(), updateAccount.autoRenewSecs(), updateAccount.autoRenewAccountNumber());
+        final var updateMeta = new ExpiryMeta(
+                op.hasExpirationTime() ? op.expirationTime().seconds() : NA,
+                op.hasAutoRenewPeriod() ? op.autoRenewPeriod().seconds() : NA,
+                NA);
+        context.expiryValidator().resolveUpdateAttempt(currentMetadata, updateMeta);
```

Comment:
Shouldn't we be saving the resolved `ExpiryMeta` and using it in what follows to complete the update?

E.g.
```
final var postUpdateMeta =
  context.expiryValidator().resolveUpdateAttempt(currentMetadata, updateMeta);

if (updateAccount.expiredAndPendingRemoval()) {
    // For an expired account, the update must extend its expiration past the current consensus time
    validateTrue(
      postUpdateMeta.expiry() > handleContext.consensusNow().getEpochSecond(), 
      ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
}
```

Reply:
As discussed, we were doing a check inside `resolveUpdateAttempt`